### PR TITLE
Possible memory leaks fix -- patch from issue #17412

### PR DESCRIPTION
**Pull request by Rubonnek**
Some apparent memory leaks were found on issue #17412.

I'm not quite familiar with the code that's being modified -- I just applied the patches created by @ka7713 and created the pull request since the patches do seem to fix a memory leak. Further review is needed.

### CODE REVIEW

File: cocos/base/CCDirector.cpp [hunk 1]
```diff
@@ -210,6 +216,13 @@ Director::~Director(void)
     
     Configuration::destroyInstance();
 
+    Sprite3DMaterial::releaseBuiltInMaterial();
+
+    cocostudio::TransformHelp::release();
+
+    ObjectFactory::destroyInstance();
```
Comment (minggo) [inline cocos/base/CCDirector.cpp:224]: cocos studio is not necessary by the game, so it may not exist.

Comment (Rubonnek) [reply cocos/base/CCDirector.cpp:224]: Added a pointer check in TransformHelp::release in the latest commit.

Comment (minggo) [reply cocos/base/CCDirector.cpp:224]: I mean developers may remove cocos studio/particle3d from project building, so `Director` can not be so tightly with these modules.

Comment (Rubonnek) [reply cocos/base/CCDirector.cpp:224]: Got it, I removed them from `Director` in the latest commit. Let me know if there's anything else that needs to be done -- I'll work on it.

**Pull request closed without merging**